src/index.ts

```diff
@@ -14,24 +15,28 @@ import {
 import { JSFOptions, JSFCallback } from './utils';
 
 export interface MiddlewareOptions {
-  file: string;
+  file: string | OpenAPIV3.Document;
+  inMemory: boolean;
```

Comment:
I think a better approach would be not to use this new option at all.
We can check the type of the `file` option. If it is a string, we can assume it's a filepath; otherwise it's a doc object.

Example: https://github.com/APIDevTools/json-schema-ref-parser/blob/main/lib/normalize-args.js#L22

Reply:
inMemory removed and spec added